service/src/main/java/com/epam/esm/service/CertificateService.java

```diff
@@ -0,0 +1,78 @@
+package com.epam.esm.service;
+
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.repository.SortType;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Interface that provides functionality for manipulating {@link Certificate} entity.
+ *
+ * @author Geller Egor
+ */
+public interface CertificateService {
+    /**
+     * Retrieve certificates according to specified parameters. All parameters are optional
+     * and can be used in conjunction, if they are not present, all certificates will be retrieved
+     *
+     * @param tagName                name of the tag
+     * @param certificateName        certificate name (part-qualified)
+     * @param certificateDescription certificate description (part-qualified)
+     * @param sortByName             sort names in ASC/DESC order
+     * @param sortByCreateDate       sort creation date in ASC/DESC order
+     * @return list of {@link Certificate}
+     */
+    List<CertificateDto> findCertificateByService(String tagName, String certificateName, String certificateDescription,
+                                                  SortType sortByName, SortType sortByCreateDate);
+
+    /**
+     * Find certificate by its id
+     *
+     * @param id certificate id
+     * @return {@link Optional} of {@link Certificate} entity.
+     */
+    CertificateDto findCertificateByIdService(long id);
+
+    /**
+     * Attach tag to existing certificate.
+     *
+     * @param certificateId certificate id
```

Comment:
If you use the JavaDoc plugin for documentation, you need to correct this "auto-generation"

Reply:
The generating is correct :)